osfoffline/views/start_screen.py

```diff
@@ -35,73 +35,20 @@ def __init__(self):
-        except (aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError, aiohttp.errors.TimeoutError):
-            # No internet connection
-            QMessageBox.warning(
-                None,
-                "Log in Failed",
-                "Unable to connect to server. Check your internet connection or try again later."
-            )
-        except (aiohttp.errors.HttpBadRequest, aiohttp.errors.BadStatusLine):
-            # Invalid credentials
-            QMessageBox.warning(
-                None,
-                "Log in Failed",
-                "Invalid login credentials."
-            )
-        else:
-            json_resp = yield from resp.json()
-            remote_user = RemoteUser(json_resp['data'])
-
-            try:
-                user = session.query(User).filter(User.osf_id == remote_user.id).one()
-            except MultipleResultsFound:
-                logging.warning('multiple users with same username. deleting all users with this username. restarting function.')
-                for user in session.query(User).filter(User.osf_id == remote_user.id).all():
-                    session.delete(user)
-                    try:
-                        save(session, user)
-                    except Exception as e:
-                        logging.exception(e)
-                        QMessageBox.warning(
-                            None,
-                            "Log in Failed",
-                            "Unable to save user data. Please try again later."
-                        )
-
-                return self.log_in()
-            except NoResultFound:
-                logging.debug('user doesnt exist. Creating user. and logging them in.')
-                user = User(
-                    full_name=remote_user.name,
-                    osf_id=remote_user.id,
-                    # osf_login='',  # TODO: email goes here when more auth methods are added, not currently returned by APIv2
-                    osf_local_folder_path='',
-                    oauth_token=personal_access_token,
-                )
-            else:
-                if not user.oauth_token == personal_access_token:
-                    user.oauth_token == personal_access_token
-
-            user.logged_in = True
-            try:
-                save(session, user)
-            except Exception as e:
-                logging.exception(e)
-                QMessageBox.warning(
-                    None,
-                    "Log in Failed",
-                    "Unable to save user data. Please try again later."
-                )
+            user = yield from auth_client.log_in(username=username, password=password)
+        except Exception as e:
+            logging.exception(e)
+            user = None
 
+        if user:
+            logging.debug('Successfully logged in user: {}'.format(user))
             self.close()
+        else:
+            logging.debug('Login Failed')
```

Comment:
Separation of concerns note: arguably, `AuthClient` should not be responsible for UI work. It should raise errors and allow the UI client decide how to handle them with an appropriate message.